### DISCUSSION TITLE
Silent install of PortableGit

### DISCUSSION THREAD

**Discussion by FlowMatric** [git-for-windows/git#5027] [General]
Hi - I am curious about your experiences installing the portable version of git silently. In my case, no matter what arguments I am passing to the installer, it always opens a dialog asking for the installation directory. 

I am working with **PortableGit-2.45.2-64-bit.7z.exe** and tried the following arguments:

> - /VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /COMPONENTS="gitlfs,ext\reg\shellhere,assoc,assoc_sh"
> - /SILENT /INSTALLDIR="C:\tmp\asdf" /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /COMPONENTS="gitlfs,ext\reg\shellhere,assoc,assoc_sh"
> - and many subsets of the arguments above...

Your ideas and any help is much appreciated!

**Comment by dscho** [comment]
These arguments work with the _installer_, but not with Portable Git. The latter is a completely different thing: not an InnoSetup-based installer, but instead a self-extracting 7-Zip archive with _very_ limited installer options. Basically, Portable Git allows you to specify _where_ to extract it to, and that's it. All other configurations need to be done manually (usually by calling `git config --system ...`).

[EDIT]: So if you want a _silent_ installation, just use 7-Zip directly to extract the archive (use it as if it did not have the extension `.7z.exe` but instead `.7z`).

**Comment by rimrul** [comment]
> no matter what arguments I am passing to the installer, it always opens a dialog asking for the installation directory.

You should be able to skip that dialog with the `-y` option. In addition you should be able to specify the installation directory with the `-o` option. There might be some graphical elements remaining, but it should work for an unattended install.

**Reply by dscho** [reply]
Indeed. And those are [the only two supported options, and they are so far undocumented](https://github.com/git-for-windows/7-Zip/blob/24ee647c64d9035be343c8402403734d9b01f5a5/CPP/7zip/Bundles/SFXSetup/SfxSetup.cpp#L168-L212). PRs welcome!
